hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ClusterStateEndpoint.java

```diff
@@ -127,13 +131,19 @@ public Response getClusterState() {
     }
 
     Long totalKeys = 0L;
+    Long deletedKeys = 0L;
+
     if (keyRecord != null) {
       totalKeys += keyRecord.getValue();
     }
     if (fileRecord != null) {
       totalKeys += fileRecord.getValue();
     }
+    if (deletedKeyRecord != null) {
+      deletedKeys += deletedKeyRecord.getValue();
+    }
     builder.setKeys(totalKeys);
```

Comment:
I think this needs to be corrected here. If keys moved to deleted table, then actual total number of keys should also be updated.

Reply:
I think , same holds true for container count. What if all keys mapped to a container are deleted, then container count also should reflect the true count. Though it is a soft delete, but when keys will be deleted, then I believe they will add an entry to deletedTable and key reference would have removed from keyTable.

Reply:
Also I think the keys shown on overview page is now support FSO bucket keys as well, so check for FSO bucket keys as well.
Its good we can refer  deletedDirectoryTable also to find out the number of file paths deleted or no longed referenced.

Reply:
IMO, we should just update Overview page data like number of deleted keys in case of LEGACY/OBS buckets and  number of deleted paths in case of FSO buckets. Details of those keys can be shown and done in new requirement where I am working currently on design for OM DB Insights and SCM DB insights page.

Reply:
> I think this needs to be corrected here. If keys moved to deleted table, then actual total number of keys should also be updated.

Yes, the client sends a delete key request to the OM. Then, the OM moves the key from the key table (visible namespace) to the deleted key table (internal). After that, the OM returns success to the client. Hence, before incrementing the count of keys for the deleted table, we will first decrement the count of keys from the total keys. I have manually checked it as well, the total key count gets decremented automatically.

Reply:
> IMO, we should just update Overview page data like number of deleted keys in case of LEGACY/OBS buckets and number of deleted paths in case of FSO buckets. Details of those keys can be shown and done in new requirement where I am working currently on design for OM DB Insights and SCM DB insights page.

Thank you for sharing your suggestion. I agree with you that it would be a good idea to differentiate the number of deleted keys in **OBS/Legacy** from **FSO**, and we can implement a separate count in the endpoint response for this purpose. However, for the user interface, I suggest we add both counts up and display them as the total number of deleted keys, which will make it easier for users to understand the overall state of the system.